staging/src/k8s.io/apiserver/pkg/storage/cacher/time_budget.go

```diff
@@ -48,43 +50,47 @@ type timeBudgetImpl struct {
 	sync.Mutex
 	budget time.Duration
 
-	refresh   time.Duration
-	maxBudget time.Duration
+	refresh         time.Duration
+	maxBudget       time.Duration
+	clock           clock.Clock
+	lastRefreshTime time.Time
 }
 
-func newTimeBudget(stopCh <-chan struct{}) timeBudget {
-	result := &timeBudgetImpl{
-		budget:    time.Duration(0),
-		refresh:   refreshPerSecond,
-		maxBudget: maxBudget,
-	}
-	go result.periodicallyRefresh(stopCh)
-	return result
-}
-
-func (t *timeBudgetImpl) periodicallyRefresh(stopCh <-chan struct{}) {
-	ticker := time.NewTicker(time.Second)
-	defer ticker.Stop()
-	for {
-		select {
-		case <-ticker.C:
-			t.Lock()
-			if t.budget = t.budget + t.refresh; t.budget > t.maxBudget {
-				t.budget = t.maxBudget
-			}
-			t.Unlock()
-		case <-stopCh:
-			return
-		}
+func newTimeBudget(clk clock.Clock) timeBudget {
```

Comment:
nit: let's try to not use abbreviations; since you can't use "clock", maybe use inputClock or sth like that

Reply:
Ah! Sure. I was apprehensive of the same. Just couldn't find a better word for it.